domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasNumeratorEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,203 @@
+package org.oppia.android.domain.classify.rules.fractioninput
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.Fraction
+import org.oppia.android.app.model.InteractionObject
+import org.oppia.android.domain.classify.RuleClassifier
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [FractionInputHasNumeratorEqualToRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputHasNumeratorEqualToRuleClassifierProviderTest {
+  private val NON_NEGATIVE_VALUE_0 = createNonNegativeInt(value = 0)
+  private val WHOLE_NUMBER_123 = createWholeNumber(isNegative = false, value = 123)
+  private val FRACTION_2_OVER_4 = createFraction(isNegative = false, numerator = 2, denominator = 4)
+  private val SIGNED_INT_1 = createSignedInt(1)
+  private val SIGNED_INT_2 = createSignedInt(2)
+  private val SIGNED_INT_NEGATIVE_2 = createSignedInt(-2)
+
+  @Inject
+  internal lateinit var fractionInputHasNumeratorEqualToRuleClassifierProvider:
+    FractionInputHasNumeratorEqualToRuleClassifierProvider
+
+  private val numeratorIsEqualClassifierProvider: RuleClassifier by lazy {
+    fractionInputHasNumeratorEqualToRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Test
+  fun testEquals_wholeNumber123Answer_withSignedInt1Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("x" to SIGNED_INT_1)
+
+    val matches =
+      numeratorIsEqualClassifierProvider.matches(
+        answer = WHOLE_NUMBER_123,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_fraction2Over4Answer_withSignedInt1Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("x" to SIGNED_INT_1)
+
+    val matches =
+      numeratorIsEqualClassifierProvider.matches(
+        answer = FRACTION_2_OVER_4,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_fraction2Over4Answer_withSignedIntNegative2Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("x" to SIGNED_INT_NEGATIVE_2)
+
+    val matches =
+      numeratorIsEqualClassifierProvider.matches(
+        answer = FRACTION_2_OVER_4,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_fraction2Over4Answer_withSignedInt2Input_bothValuesMatch() {
```

Comment:
Other possible tests that could be added:
- 1/2 and 2 do not match (verifying that denominators are not matched here)
- 1/2 and 2/4 do not match since although they're equal, the numerators do not match
- -2 and -2/4 do match (negative numerator equals)
- 1/2 and 1/2 match (identity check)

Reply:
The matcher takes in a SIGNED_INT as the input and FRACTION as answer, and compares the SIGNED_INT with the numerator of the FRACTION, so I think some of these cases don't apply for this test suite. I added a case for a negative signed int and negative numerator.

Reply:
Ah, yes. I'm not quite sure what I was thinking when suggesting to use fraction inputs. :) Good call.